<a href="https://colab.research.google.com/github/om4subhash/python/blob/Week2/Week_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Variable
ram_bank_balance = 100000
ram_loan = 500000

lakshman_bank_balance = 200000
lakshman_loan = 1000000

net_income = ram_bank_balance + lakshman_bank_balance
net_liablity = ram_loan + lakshman_loan

final_value = net_income - net_liablity

print("So, the family has",final_value)

So, the family has -1200000


Variable: Dynamic Typing

In [ ]:
a = 10
print(type(a))

a = 'India'
print(type(a))

n = 10
print(type(n))
n = 10/2
print(type(n))
print(n)

<class 'int'>
<class 'str'>
<class 'int'>
<class 'float'>
5.0


keyword: is predefined in python. so can't use as variable name.

Variable, Operator and Expression
keyword: is predefined in python. so can't use as variable name.
Variable- only use alphanumerical character, only one special character _
eg: a1, a_ are the valid variable name
must start with alphabate/number/_
eh: _1a
case sensative
roll =5
ROLL = 10
Roll = 15
Multiple assignment:
x,y = 1,2
print(x,y)
seuence matter so be cairefule
x=y=z = 10
print(x,y,z)
------------
Deleting the variable

x= 10
print(x)
del x
print(x) #error: name 'x' is not defined

short hand operators: remove the repeatative variable
use with all arethematic operation

count = 0
count = count + 1
count = count + 1
count = count + 1
print(count)
count = 0
count  += 1
count  += 1
count  += 1
print(count)
---------------------
Special operator
print('alpha' in 'A variable can be alpha-numerical')
print('alpha' in 'A variable can be start with underscore or name')
-----------------------------
Chaning Operatrion: Multiple rellational operation in single statement is called chaning operation

x = 5
print(1 < x < 10)
print(10 < x < 20)
print(x < 10 < x * 10 < 100)
print(10 > x <= 9)
print(5 == x > 4)

In [ ]:
roll =5
ROLL = 10
Roll = 15
print(roll)
print(ROLL)
print(Roll)

x,y = 1,2
print(x,y)

x=y=z = 20
print(x,y,z)

x= 10
print(x)
del x
#print(x)

count = 0
count = count + 1
count = count + 1
count = count + 1
print(count)
count = 0
count  += 1
count  += 1
count  += 1
print(count)

count = 10
count -= 1
count = count -1
print(count)

count = 2
count *= 2
count = count * 2
print(count)

print('alpha' in 'A variable can be alpha-numerical')
print('alpha' in 'A variable can be start with underscore or name')

x = 5
print(1 < x < 10)
print(10 < x < 20)
print(x < 10 < x * 10 < 100)
print(10 > x <= 9)
print(5 == x > 4)

5
10
15
1 2
20 20 20
10
3
3
8
8
True
False
True
False
True
True
True


Escape Characters and types of quotes
allows us to use specefil symbole in code, which is not use normally
print('It's beautyful day')
error:     print('It's beautyful day') #error
                             ^
SyntaxError: unterminated string literal (detected at line 1)

print('It\'s beautyful day')
-------------------
Use of court supported by python:

z = 'frist line
second line
thierd line'
single court use for only single line, so use '''

In [ ]:
#print('It's beautyful day') #error
print('It\'s beautyful day')
#print("We are form "IIT" Madras")
print("We are form \"IIT\" Madras")
print('My name is Om Subhash. I am from Bihar')
print("My name is Om Subhash.\t I am from Bihar")
print("My name is Om Subhash.\n I am from Bihar")

print("It's beautyful day")
print('We are form "IIT" Madras')

x = 'This is a string'
y = 'This is also a string'
#z = 'frist line
#second line
#thierd line'
z = '''frist line
#second line
#thierd line'''
print(x)
print(y)
print(z)

It's beautyful day
We are form "IIT" Madras
My name is Om Subhash. I am from Bihar
My name is Om Subhash.	 I am from Bihar
My name is Om Subhash.
 I am from Bihar
It's beautyful day
We are form "IIT" Madras
This is a string
This is also a string
frist line
#second line
#thierd line


String Method
lower()--> converts the string into lower case. print(x.lower())
upper()--> converts the stringh into upper case. print(x.upper())
capitalize() --> converts the first character to upper case. print(x.capitalize())
title()--> convert the first character of the each letter into upper case. print(x.title())
sqapcase()--> Swaps csases, lower case becomes upper case and vice versa. print(x.swapcase())
islower()--> Return True if all characters in the string are lower case. print(x.islower())
isupper()--> Return True if all the characters in the string are upper case.
print(x.isupper())
istitle()--> Returns True if the string follows the rules of the title.
print(x.istitle())
isdigit()--> Returns True if all characters in the string are digits.
print(x.isdigit())
isalpha()--> Returns True if all characters in the string are in alphabets.
print(x.isalpha())
isalnm()--> Return True if all characters in the string are alpha-numeric
print(x.isalnum())
strip()--> Return a trimmed version of the string. print(x.strip('-'))
lstrip()--> Return a left trim version of the string. print(x.lstrip('-'))
rstrip--> Return a right trim version of the string. print(x.rstrip('-'))
startwith()--> Return True if the string starts with the specified value.
print(x.startwith('P'))
endwith()--> Return True if the string ends with the specified value.
print(x.endwith('N'))
count()--> Retrun the cumber of times a specified value occures in a string.
print(x.count())
index()--> Searchs the string for a specified value and returns the position of where it was found. print(x.index())
replace()--> Returns a string where a specifies value is replaced with a specifies value.  print(x.replace('S','a'))

In [ ]:
# Ceasercipher in cryptography
alpha = 'abcdefghijklmnopqrstuvwxyz'
'''i = 20
print(alpha[i])
print(alpha[i+1])
print(alpha[i+2])
print(alpha[i+3])
i = 25
print(alpha[i])
#print(alpha[i+1])
#print(alpha[i+2])
#print(alpha[i+3])
i = 30
print(alpha[i%26])
i=26
print(alpha[i%26])
i = 24
print(alpha[i%26])
print(alpha[(i+1)%26])
print(alpha[(i+2)%26])'''

s = 'india'
#i expect to output sift by 1
t = ''
i = 0
k = 2
t = t+(alpha[(alpha.index(s[i])+k)%26])
t = t+(alpha[(alpha.index(s[i+1])+k)%26])
t = t+(alpha[(alpha.index(s[i+2])+k)%26])
t = t+(alpha[(alpha.index(s[i+3])+k)%26])
t = t+(alpha[(alpha.index(s[i+4])+k)%26])

print(t)


kpfkc


Introduction to the If statement

In [ ]:
# Let us consider the movie "Avengers". This is a 13+ movie.
print("Please enter your date of birth")
birth_year = int(input())
current_year = 2024

age = current_year - birth_year
if (age < 13):
  print("You are under age, you can not watch this movie")
  print("Wait until you are old enough to watch this movie.")
else:
  print("you ara old enough to watch Avenges, enjoy!")
  print("Dont forget to watch the sequal and prequels.")

print("Have a nice time")

Please enter your date of birth
2000
you ara old enough to watch Avenges, enjoy!
Dont forget to watch the sequal and prequales
Have a nice time


Find whether the given number is even or odd.

In [ ]:
number = int(input('Enter the natural number :'))

if(number % 2 == 0):
  print('Even Number')
else:
  print('Odd Number')

Enter the natural number :-10
Even Number


Find whether the given number ends with 0 or 5 or any other number

In [ ]:
number = int(input('Enter the number :'))

if(number % 10 == 0):
  print('Number ends with 0')
elif(number % 5 == 0):
  print('Number ends with 5')
else:
  print("Other")

Enter the number :42
Other


Find the grade of student based on the given marks from 0 to 100

In [ ]:
marks = int(input('Enter the marks  :'))
if (marks >= 0 and marks <= 100):
  if(marks >= 90):
    print('Grade A')
  elif(marks >= 80 and marks <90):
    print('Grade B')
  elif(marks >=70 and marks < 80):
    print('Grade C')
  elif(marks >= 60 and marks < 70):
    print('Grade D')
  else:
    print('Grade E')
else:
  print('Invalid Marks')

Enter the marks  :-7
Invalid Marks


Conver the flow chart into python code

In [ ]:
print('Travel from city A to city B')
time = int(input('Enter the time :'))
longer = int(input('Define longer :'))
if(time >= longer):
  price = int(input('Enter Price :'))
  higher = int(input('Define higher :'))
  if(price >= higher):
    print('Train')
  else:
    print('Coach')
else:
  price = int(input('Enter Price :'))
  higher = int(input('Define Higher :'))
  if(price >= higher):
    print('Day Time Flight')
  else:
    print('Red Eye Flight')


Travel from city A to city B
Enter the time :2
Define longer :1
Enter Price :1000
Define higher :1000
Train


#import libery


In [ ]:
import math
#math is the liberary
print(math.log(10))
print(math.sin(90))
print(math.sqrt(16))
print(math.factorial(5))
print(math.pow(10,3))

2.302585092994046
0.8939966636005579
4.0
120
1000.0


In [ ]:
import random  #every time you get random number between 0 and 1
print(random.random())

0.8537774970515083


In [ ]:
#Let us simulate a coin toss
import random

a = random.random()

if (a < .5):
  print('Head')
else:
  print('Tail')

Head


In [ ]:
#Let us simulate a dice(six faced)

import random
dice1 = (random.randrange(1,7))
dice2 = (random.randrange(1,7))
total = dice1 + dice2
print(total)


10


In [ ]:
import calendar
#bring entire libery

print(calendar.month(2021,7))  #dispaly one month
print(calendar.calendar(2021))  #display entire year


     July 2021
Mo Tu We Th Fr Sa Su
          1  2  3  4
 5  6  7  8  9 10 11
12 13 14 15 16 17 18
19 20 21 22 23 24 25
26 27 28 29 30 31

                                  2021

      January                   February                   March
Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su
             1  2  3       1  2  3  4  5  6  7       1  2  3  4  5  6  7
 4  5  6  7  8  9 10       8  9 10 11 12 13 14       8  9 10 11 12 13 14
11 12 13 14 15 16 17      15 16 17 18 19 20 21      15 16 17 18 19 20 21
18 19 20 21 22 23 24      22 23 24 25 26 27 28      22 23 24 25 26 27 28
25 26 27 28 29 30 31                                29 30 31

       April                      May                       June
Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su
          1  2  3  4                      1  2          1  2  3  4  5  6
 5  6  7  8  9 10 11       3  4  5  6  7  8  9       7  8  9 10 11 12 13
12 13 14 15 16 17 18      10 11 12 13 14 15 1

In [ ]:
from calendar import *
#bring every thing into python program
print(calendar(2021))
print(month(2021,10))



                                  2021

      January                   February                   March
Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su
             1  2  3       1  2  3  4  5  6  7       1  2  3  4  5  6  7
 4  5  6  7  8  9 10       8  9 10 11 12 13 14       8  9 10 11 12 13 14
11 12 13 14 15 16 17      15 16 17 18 19 20 21      15 16 17 18 19 20 21
18 19 20 21 22 23 24      22 23 24 25 26 27 28      22 23 24 25 26 27 28
25 26 27 28 29 30 31                                29 30 31

       April                      May                       June
Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su      Mo Tu We Th Fr Sa Su
          1  2  3  4                      1  2          1  2  3  4  5  6
 5  6  7  8  9 10 11       3  4  5  6  7  8  9       7  8  9 10 11 12 13
12 13 14 15 16 17 18      10 11 12 13 14 15 16      14 15 16 17 18 19 20
19 20 21 22 23 24 25      17 18 19 20 21 22 23      21 22 23 24 25 26 27
26 27 28 29 30            24 25 26 27 

In [ ]:
#import specific feature from the liberary and store in varaible
import calendar as c
from calendar import month as m

x is the varable of type float and is of the form a.bcd, where a, b, c, d are all positive integers less than 10. what is the output of the following snipppet of code?

a.bcd means==> it is the float format

In [ ]:
E_1 = False
E_2 = False
E_3 = False
if E_1:
  x = 1
if E_2:
  x = 2
if E_3:
  x = 3
print(x)

NameError: name 'x' is not defined

In [ ]:
a,b,c,d = input()
print(a)
print(b)
print(c)
print(d)

1234
1
2
3
4


In [ ]:
word = input()
valid = False
if 'a' <= word[0] <= 'z':
#word[0] within the ASCI range of lower case
  if word[0] == word[-1]:
    valid = True
print(valid)

asDa
True


In [ ]:
word = input()
match = False
if word.count('(') == word.count(')'):
  if word.count('{') == word.count('}'):
    if word.count('[') == word.count('}'):
      match = True

if match:
  print('Perfect!')
else:
  print('IMPERFECT')

abcd
word abcd
Perfect!


Accept a non-zero integer as input.Print positive if it is grater thane zero and negative if it is less than zero.

In [ ]:
no = int(input())
if no > 0:
  print('Positive')
else:
  print('Negative')

0
Negative


In [ ]:
x = int(input())

if 0 < x < 10:
  print(x + 2)
elif x <= 10:
  print((((x)**2)) + 1)
else:
  print(0.0)

100
0


Accept an integer as input and print the time of the day. Use the following table for reference.
Input
Output
T< 0
INVALID
O<T<5
NIGHT
6≤T ≤11
MORNING
12 ≤ T ≤ 17
AFTERNOON
18 ≤ T ≤ 23
EVENING
T ≥ 24
INVALID The input will be a single line containing an integer. The output should be one of these strings: NIGHT,
MORNING, AFTERNOON, EVENING, INVALID.

In [ ]:
T = int(input())

if 0 <= T <= 5:
  print('Night')
elif 6 <= T <= 11:
  print('Morning')
elif 12 <= T <= 17:
  print('Afternoon')
elif 18 <= T <= 23:
  print('Evening')
else:
  print('Invalid')


26
Invalid


Note: This Assignment will be considered for the Bonus Marks. The final submission before August 6th only considered.

Accept a point in 2D space as input and find the region in space that this point belongs to. A point could belong to one of the four quadrants, or it could be on one of the two axes, or it could be the origin. The input is given in 2 lines: the first line is the x-coordinate of the point while the second line is its y-coordinate. The possible outputs are first, second, third, fourth, x-axis, y-axis, and origin. Any other output will not be accepted. Note that all outputs should be in lower case.

In [ ]:
x = float(input())
y = float(input())

if x == 0 and y == 0:
  print('Origin')
elif x > 0 and y > 0:
  print('First Quadrants')
elif x < 0 and y > 0:
  print('Second Quadrants')
elif x < 0 and y < 0:
  print('Third Quadrants')
elif x > 0 and y < 0:
  print('Fourth Quadrants')
elif x == 0 and y != 0:
  print('Y-Axis')
else:
  print('X-Axis')

1
-9
Fourth Quadrants


Write a program to realize the equation of a line given 2 points
(
x
1
,
y
1
)
(x
1
​
 ,y
1
​
 ) and
(
x
2
,
y
2
)
(x
2
​
 ,y
2
​
 ) in 2D space. The input is in 5 lines where, the first, second, third, and fourth lines represent
x
1
x
1
​
 ,
y
1
y
1
​
 ,
x
2
x
2
​
 , and
y
2
y
2
​
  respectively. The fifth line corresponds to
x
3
x
3
​
 . Determine
y
3
y
3
​
  using the equation of a straight line as given below:

x
−
x
1
x
2
−
x
1
=
y
−
y
1
y
2
−
y
1
x
2
​
 −x
1
​

x−x
1
​

​
 =
y
2
​
 −y
1
​

y−y
1
​

​

The output should be "Vertical Line" if the line is vertical. In other cases, the output should be 2 lined, where the first line is the value of
y
3
y
3
​
  and the second line indicates whether the slope of the line is positive, negative or zero. Print "Positive Slope", "Negative Slope" or "Horizontal Line" accordingly.

In [ ]:
x1 = float(input())
y1 = float(input())
x2 = float(input())
y2 = float(input())
x3 = float(input())



if x1 != x2:

  m = (y2 - y1)/(x2 - x1)
  y3 = m*(x3 - x1) + y1
  print(y3)

  if m > 0:
    print('Positive Slope')
  elif m < 0:
   print('Negative Slope')
  else:
    print('Horizontal Line')
else:
  print('Verticale Line')

1
2
1
6
5
Verticale Line


Accept a string as input. If the input string is of odd length, then continue with it. If the input string is of even length, make the string of odd length by following the two steps mentioned below:
(1) If the last character is a period (.), then remove it
(2) If the last character is not a period, then add a period (.) to the end of the string

Call this string of odd length word. Select a substring made up of three consecutive characters from word such that there are an equal number of characters to the left and right of this substring. Print this substring as output. You can assume that all input strings will be in lower case and will have a length of at least four.


In [ ]:
word = input()


length = len(word)


if length % 2 == 0:
  if word[-1] == '.':
    word = word[:-1]
  else:
    word = word + '.'


mid = len(word) // 2
sub_str = word[mid-1:mid+2]
print(sub_str)

hello
ell


A sequence of five words is called magical if the
i
t
h
i
th
  word is a substring of the
(
i
+
1
)
t
h
(i+1)
th
  word for every
i
i in the range
1
≤
i
<
5
1≤i<5. Accept a sequence of five words as input, one word on each line. Print magical if the sequence is magical and non-magical otherwise.

Note that str_1 is a substring of str_2 if and only if str_1 is present as a sequence of consecutive characters in str_2.



In [ ]:
word1 = input()
word2 = input()
word3 = input()
word4 = input()
word5 = input()

count = 0

if word1 in word1:
  count += 1
if word2 in word3:
  count += 1
if word3 in word3:
  count += 1
if word4 in word5:
  count += 1
if count == 4:
  print('Magical')
else:
  print('Non-Magical')

a
aa
aaa
aaaa
aaaaa
Non-Magical


Accept two positions as input: start and end. Print YES if a bishop at start can move to end in exactly one move. Print NO otherwise. Note that a bishop can only move along diagonals.
The algorithm for this problem is given as part of the template code.Implement the algorithm in Python.

In [ ]:
start = input()
end = input()
h_position = 'ABCDEFGH'

start_hor = start[0]
end_hor = end[0]
start_var = int(start[1])
end_var = int(end[1])

start_hor_index = h_position.index(start_hor)
end_hor_index = h_position.index(end_hor)

if abs(start_hor_index - end_hor_index) == abs(start_var - end_var):
  print('Yes')
else:
  print('No')

C3
E5
Yes


You have
n
n gold coins with you. You wish to divide this among three of your friends under the following conditions:
(1) All three of them should get a non-zero share.
(2) No two of them should get the same number of coins.
(3) You should not have any coins with you at the end of this sharing process.
The input has four lines. The first line contains the number of coins with you. The next three lines will have the share given to your three friends. All inputs shall be non-negative integers. If the division satisfies these conditions, then print the string FAIR. If not, print UNFAIR.

In [ ]:
# CODE WITH ALGORITHM
# Step 1: get the first line as an integer 'n'
n = int(input())
# Step 2: get the next three lines as integers as share1, share2, and share3
share1 = int(input())
share2 = int(input())
share3 = int(input())
# Step 3: initialize a boolean variable is_fair as True.
is_fair = True
# Step 4: Check for opposites of the given three conditions one by one.
if (share1 or share2 or share3) != 0:
  if n == (share1 + share2 + share3):
    if share1 != share2 and share2 != share3 and share3 != share1:
      is_fair = True
    else:
      is_fair = False
  else:
    is_fair = False
else:
  is_fair = False

if is_fair:
  print('FAIR')
else:
  print('UNFAIR')
# Step 5: If the opposite of the condition is true then set is_fair to False.
# Stpe 6: if is_fair is true print "FAIR" else print "UNFAIR"

6
2
2
2
UNFAIR


In [ ]:
# CODE WITH ALGORITHM
# Step 1: get the first line as an integer 'n'
n = int(input())
# Step 2: get the next three lines as integers as share1, share2, and share3
share1 = int(input())
share2 = int(input())
share3 = int(input())
# Step 3: initialize a boolean variable is_fair as True.
is_fair = True
# Step 4: Check for opposites of the given three conditions one by one.
if share1 == 0 or share2 == 0 or share3 == 0:
  is_fair = False

if share1 == share2 or share1 == share2 or share2 == share3:
  is_fair = False

if n != (share1 + share2 + share3):
  is_fair = False
# Step 5: If the opposite of the condition is true then set is_fair to False.
# Stpe 6: if is_fair is true print "FAIR" else print "UNFAIR"

if is_fair:
  print('FAIR')
else:
  print('UNFAIR')

6
2
4
0
UNFAIR


Accept a real number
x as input and print the greatest integer less than or equal to
x on the first line, followed by the smallest integer greater than or equal to x on the second line.

In [ ]:
no = float(input())

isFloat = isinstance(no, float)

no1 = 0
no2 = 0
round_no = 0

if isFloat:
  round_no = round(no)

  if no < 0 and round_no > no:
    no1 = round_no - 1
    no2 = round_no
  elif round_no > no:

    no1 = round_no - 1
    no2 = round_no
  else:

    no1 = round_no
    no2 = round_no + 1

else:
  no1 = no
  no2 = no

print((no1))
print((no2))


10.3
10
11


In [ ]:
no = float(input())

if no > 0:
  floor = int(no)
  if floor == no:
    ceil = floor
  else:
    ceil = floor + 1

else:

  ceil = int(no)
  if ceil == no:
    floor = ceil
  else:
    floor = ceil - 1

print(floor)
print(ceil)

5
5
5


In [ ]:
no = int(input())

rem = no % 4

if rem == 0:
  print('Emerald')
elif rem == 3:
  print('Ruby')
elif rem == 2:
  print('Peridot')
else:
  print('Sapphire')


222
Peridot


A data science company wants to hire data scientists from IIT Madras. The company follows a certain criteria for selection: for a student to be selected, the number of backlogs should be at most 5 and the CGPA (Cumulative Grade Point Average) should be greater than 6. If the student does not fit the above criteria, then the student is not offered the job. If the student is selected, then the salary offered is equal to 5 times his/her CGPA (in lakhs).

Accept the number of backlogs (integer) and the CGPA (float) of the student as input. Your task is to determine if the student is selected or not. If the student is selected, then print the package. If not, then print the string Not Selected.

In [ ]:
backlogs = int(input())
cgpa = float(input())

is_select = False
if backlogs <= 5 and cgpa >= 6:
  is_select = True
else:
  is_select = False

if is_select:
  print( cgpa * 5 )
else:
  print('Not Selected')

30.0


In [ ]:
backlog = int(input())
cgpa = float(input())

if backlog > 5 or cgpa <= 6:
  print('Not Selected')
else:
  print(cgpa * 5)

5
6
Not Selected


A test match happened between Team A and Team B. The scores of the teams in both the innings are given as input in eight lines in the format given below. The first two lines represent the scores of Team A in the first innings and the next two lines represent the scores of Team A in the second innings. Likewise, the last four lines represent the scores of Team B in both the innings.

The numbers in 2nd, 4th, 6th, and 8th lines represent the wickets lost by the teams and the numbers in 1st, 3rd, 5th, and 7th represent the runs scored. For example:

120
10
210
10
115
10
189
10

In the above example, team-A has scored
120
120 for the loss of
10
10 wickets in the first innings, and
210
210 for the loss of
10
10 wickets in the second innings. Team A plays first and Team B plays second. Your task is to determine the winner of the match.

Process to decide the outcome

Team A wins if and only if the sum of its scores in both the innings is greater than sum of the scores of Team B in both the innings AND Team B lost all the ten wickets in the second innings. Team B wins if the sum of its scores in both the innings is greater than sum of the scores of Team A in both the innings.

A match will end in a draw if the sum of scores in the two innings of both the teams are equal OR if Team B did not lose all the ten wickets in the second innings. If the match ends in a draw, then print DRAW.

Example

120
10
210
10
115
10
189
10
Example output

Team A

120
+
210
>
115
+
89
120+210>115+89 and Team B lost all
10
10 wickets in second innings, therefore Team A is the winner of the test match.

In [ ]:
A_1_runs, A_1_wickets, A_2_runs, A_2_wickets = int(input()), int(input()), int(input()), int(input())
B_1_runs, B_1_wickets, B_2_runs, B_2_wickets = int(input()), int(input()), int(input()), int(input())
total_A_runs = A_1_runs + A_2_runs
total_B_runs = B_1_runs + B_2_runs

if(total_B_runs > total_A_runs):
    print("Team B")
elif(total_A_runs > total_B_runs and B_2_wickets == 10):
    print("Team A")
else:
    print("DRAW")

A word is said to be perfect if it satisfies all the following criteria:

(1) All the vowels (a,e,i,o,u) should be present in the word.

(2) Let the vowels be represented as
v
1
=
a
,
v
2
=
e
,
v
3
=
i
,
v
4
=
o
,
v
5
=
u
v
1
​
 =a,v
2
​
 =e,v
3
​
 =i,v
4
​
 =o,v
5
​
 =u in lexical order.

If
i
<
j
i<j, then the first appearance of
v
i
v
i
​
  in the word should come before the first appearance of
v
j
v
j
​
 .
If
i
<
j
i<j, then the count of
v
i
v
i
​
  should be greater than or equal to the count of
v
j
v
j
​
 .
Accept a word as input and do the following:
 (1) If it is a perfect word, print It is a perfect word
 (2) If the word is not a perfect word, print It is not a perfect word

In [ ]:
word = input()
present = False
if('a' in word and 'e' in word and 'i' in word and 'o' in word and 'u' in word):
    if (word.index('a') < word.index('e') < word.index('i') < word.index('o') < word.index('u')):
        if(word.count('a') >= word.count('e') >= word.count('i') >= word.count('o') >= word.count('u')):
            present = True
if (present):
    print ('It is a perfect word')
else:
    print('It is not a perfect word')

aeiuo
It is not a perfect word


Accept four integers as input and write a program to print these integers in non-decreasing order.

The input will be four integers in four lines. The output should be a single line with all the integers separated by a single space in non-decreasing order.

Note: There is no space after the fourth integer.

In [ ]:
a = int(input())
b = int(input())
c = int(input())
d = int(input())

if (a <= b and a <= c and a <= d):
    if(b <= c and b <= d):
        if(c <= d):
            print(a,b,c,d)
        else:
            print(a,b,d,c)
    elif(c <= b and c <= d):
        if(b <= d):
            print(a,c,b,d)
        else:
            print(a,c,d,b)
    else:
        if(b <= c):
            print(a,d,b,c)
        else:
            print(a,d,c,b)
elif (b <= a and b <= c and b <= d):
    if(a <= c and b <= d):
        if(c <= d):
            print(b,a,c,d)
        else:
            print(b,a,d,c)
    elif(c <=a and c <= d):
        if(a <= d):
            print(b,c,a,d)
        else:
            print(b,c,d,a)
    else:
        if(a <= c):
            print(b,d,a,c)
        else:
            print(b,d,c,a)
elif (c <= a and c <=b and c<= d):
    if(a <= b and a <= d):
        if(b <= d):
            print(c,a,b,d)
        else:
            print(c,a,d,b)
    elif(b <= a and b <= d):
      if(a <= d):
          print(c,b,a,d)
      else:
          print(c,b,d,a)
    else:
      if(a <= b):
          print(c,d,a,b)
      else:
          print(c,d,b,a)
else:
    if(a <= b and a <= c):
        if(b <= c):
            print(d,a,b,c)
        else:
            print(d,a,c,b)
    elif(b <= a and b <= c):
      if(a <= c):
          print(d,b,a,c)
      else:
          print(d,b,c,a)
    else:
      if(a <= b):
          print(d,c,a,b)
      else:
          print(d,c,b,a)

123
0
8988
7
0 123 7 8988


Accept three positive integers as input and check if they form the sides of a right triangle. Print YES if they form one, and NO if they do not. The input will have three lines, with one integer on each line. The output should be a single line containing one of these two strings: YES or NO.

In [3]:
lenth = int(input())
base = int(input())
hipotanous = int(input())

h = hipotanous ** 2
b = base ** 2
l = lenth ** 2

if h == (b + l):
  print("YES")
else:
  print('NO')

5
3
4
NO


In [4]:
x = int(input())
y = int(input())
z = int(input())

if ((x ** 2 + y ** 2 == z ** 2) or
    (y ** 2 + z ** 2 == x ** 2) or
    (z ** 2 + x ** 2 == y ** 2)):
    print("YES")
else:
    print("NO")

5
3
4
YES


The employees follow a strange convention. They will continue the meeting only if the following condition is satisfied.

The sum of the employee-ids of every pair of adjacent employees at the table must be an even number.

They are so lazy that they won't move around to satisfy the above condition. If the current seating plan doesn't satisfy the condition, the meeting will be canceled. You are given the employee-id of all five employees. Your task is to decide if the meeting happened or not.

The input will be five lined, each line containing an integer. The
line will have the employee-id of Ei. The output will be a single line containing one of these two strings: YES or NO.

In [7]:
e1 = int(input())
e2 = int(input())
e3 = int(input())
e4 = int(input())
e5 = int(input())

if (e1 + e2) != 0:
  print('NO')
elif(e2 + e3) != 0:
  print('NO')
elif(e3 + e4) != 0:
  print('NO')
elif(e4 + e5) != 0:
  print('NO')
else:
  print('YES')

2
4
6
8
10
NO


Accept a string as input and print the vowels present in the string in alphabetical order. If the string doesn't contain any vowels, then print the string none as output. Each vowel that appears in the input string — irrespective of its case — should appear just once in lower case in the output.

In [13]:
string = input()
vowel = ''

if 'a' in string.lower():
  vowel += 'a'
if 'e' in string.lower():
  vowel += 'e'
if 'i' in string.lower():
  vowel += 'i'
if 'o' in string.lower():
  vowel += 'o'
if 'u' in string.lower():
  vowel += 'u'

if vowel == '':
  print('None')
else:
  print(vowel)



1 YARD (yd) = 3 FEET (ft).
ae


You are given the dates of birth of two persons, not necessarily from the same family. Your task is to find the younger of the two. If both of them share the same date of birth, then the younger of the two is assumed to be that person whose name comes first in alphabetical order (names will follow Python's capitalize case format).

The input will have four lines. The first two lines correspond to the first person, while the last two lines correspond to the second person. For each person, the first line corresponds to the name and the second line corresponds to the date of birth in DD-MM-YYYY format. Your output should be the name of the younger of the two.

In [ ]:
name1 = input()
dob1 = input()
name2 = input()
dob2 = input()

dob_1 = dob1[0:2]+dob1[3:5]+dob1[6:11]
dob_2 = dob2[0:2]+dob1[3:5]+dob1[6:11]

dob_1_sum = int(dob_1[0]) + int(dob_1[1]) + int(dob_1[2])
+ int(dob_1[3]) + int(dob_1[4]) + int(dob_1[5])
+ int(dob_1[6]) + int(dob_1[7])


dob_2_sum = int(dob_2[0]) + int(dob_2[1]) + int(dob_2[2])
+ int(dob_2[3]) + int(dob_2[4]) + int(dob_2[5])
+ int(dob_2[6]) + int(dob_2[7])


if dob_1_sum == dob_2_sum:
    if name1 > name2:
        print(name1)
    else:
        print(name2)
elif dob_1_sum > dob_2_sum:
    print(name2)
else:
    print(name1)

In [15]:
name1 = input()
dob1 = input()
name2 = input()
dob2 = input()

dob_1 = dob1[0:2]+dob1[3:5]+dob1[6:11]
dob_2 = dob2[0:2]+dob1[3:5]+dob1[6:11]

if dob_1 == dob_2:
  if name1 > name2:
        print(name1)
  else:
        print(name2)
elif dob_1 > dob_2:
    print(name2)
else:
    print(name1)

hhhhkjk
10-08-1990
ashj
12-09-2000
hhhhkjk


Accept a string as input. Your task is to determine if the input string is a valid password or not. For a string to be a valid password, it must satisfy all the conditions given below:

(1) It should have at least 8 and at most 32 characters

(2) It should start with an uppercase or lowercase letter

(3) It should not have any of these characters: / \ = ' "

(4) It should not have spaces

It could have any character that is not mentioned in the list of characters to be avoided (points 3 and 4). Output True if the string forms a valid password and False otherwise.

In [21]:
password = input()

lenth = len(password)
if password[0].isalpha:
  if(lenth >= 8 and lenth <= 32):
    if(password.isupper or password.islower):
     if(('/' in password) or  ('\'' in password) or ('\\' in password) or ('\"' in password)):
        if(password.isspace):
          print(True)
        else:
          print(False)


passou1234
True
